### PR TITLE
Vaccination Details frontend

### LINKED ISSUE #7876: Migrate to a New Model for Storing Vaccination Details of a Patient
Currently CARE only supports storage of Vaccination details related to COVID. We need to generalise this and create a new model to store all vaccination details of a patient. Thereby removing these fields from the PatientRegistration Table. The Vaccination model should store comprehensive vaccination information to track a patient’s vaccination history effectively.

- [ ] Backend Changes to create the new model PatientVaccination & related viewsets and serializers with Suggestive Fields:
	•	patient (ForeignKey to PatientRegistration)
	•	vaccine_name (choices from vaccineEnum)
	•	date_of_vaccination (DateField)
	•	dose_number (PositiveIntegerField, default 1)
	•	vaccination_center (CharField, max_length=255)
	•	batch_number (CharField, max_length=100)
	•	created_by (ForeignKey to User)
	•	last_edited_by (ForeignKey to User)
- [ ] Create Mockups/Suggestion for handling the same in the UI.
- [ ] Frontend Changes in PatientRegistration Form, Patient Profile, Link in Patient Dashboard for Vaccination Details

### PR DESCRIPTION
**Pull request by hrit2773**
## Proposed Changes

- Fixes #7876

### CODE REVIEW

File: src/Components/Patient/PatientRegister.tsx [hunk 1]
```diff
@@ -228,6 +229,19 @@ export const PatientRegister = (props: PatientRegisterProps) => {
...
+    if (res?.ok && data?.results) {
+      setVaccinesList(data.results);
+    }
+    setIsVaccineLoading(false);
+    console.log(data?.results);
+    return data?.results || [];
+  }, []);
+  useEffect(() => {
+    fetchAllVaccines();
+  }, []);
```
Comment (khavinshankar) [inline src/Components/Patient/PatientRegister.tsx]: use useQuery which keeps the code clean and no need of extra states like vaccinesList and isVaccineLoading

File: src/Components/Patient/PatientRegister.tsx [hunk 2]
```diff
@@ -756,6 +777,7 @@ export const PatientRegister = (props: PatientRegisterProps) => {
       is_active: true,
       ration_card_category: formData.ration_card_category,
     };
+    console.log(data);
```
Comment (khavinshankar) [inline src/Components/Patient/PatientRegister.tsx]:
```suggestion
```

File: src/Components/Patient/PatientRegister.tsx [hunk 3]
```diff
@@ -1738,15 +1760,15 @@ export const PatientRegister = (props: PatientRegisterProps) => {
                           }
                           title={
                             <h1 className="text-left text-xl font-bold text-purple-500">
-                              COVID Details
+                              Complete Vaccination Details
```
Comment (khavinshankar) [inline src/Components/Patient/PatientRegister.tsx]: use internationalization, we are working on adding it through out the application, so when you come across an instance where it is not used while making changes, please feel free to add

File: src/Components/Patient/PatientVaccinationDetails.tsx [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+            <th className=" p-3 text-start text-lg font-bold">
+              Number of Doses
+            </th>
+            <th className=" p-3 text-start text-lg font-bold">
+              Vaccination Center
+            </th>
+          </tr>
+        </thead>
+        <tbody>
+          {props.vaccineData?.map((detail: any) => (
```
Comment (khavinshankar) [inline src/Components/Patient/PatientVaccinationDetails.tsx]: avoid using any

File: src/Components/VaccinationDetails/AddVaccinationDetails.tsx [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+    setState((prev: any) => {
+      return {
+        ...prev,
+        [e.name]: e.value,
+      };
+    });
+  };
+  const hasError = props.disallowed.includes(state.vaccine_name);
+  return (
+    <div className=" flex flex-col gap-2">
```
Comment (khavinshankar) [inline src/Components/VaccinationDetails/AddVaccinationDetails.tsx]: use form component and structure it similar to the existing forms, checkout patient register form for reference.

**Review by khavinshankar — CHANGES_REQUESTED** [review]
@hrit2773 nicely done, but few changes are needed, here are the gist of changes needed:
1. remove unnecessary console logs
2. avoid using any or unknown
3. use useQuery to fetch data
4. use translation whereever possible
5. follow the existing conversion used in care while creating new forms

**Comment by hrit2773** [comment]
@khavinshankar thanks for pointing them will make these changes and update it soon.

**Comment by hrit2773** [comment]
@khavinshankar the add vaccination details form is similar to that of consultation diagnosis. The only difference being that it has multiple form fields. we dont need the restore button offered by the form component. And when we press add vaccination details it should just display the details and not submit it by whole. Submitting will be handled by patient register. So should i still use the form component? because pressing the submit button is reloading the page. Moreover, its not a seperate form, the vaccination details are entered in the patient register form itself. Multiple vaccines can be added. So i think form component is not needed. That may make it look like a completely independent form layout

**Pull request closed without merging**